pkg/util/lsof/lsof_linux.go

```diff
@@ -0,0 +1,383 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package lsof
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"syscall"
+
+	"github.com/prometheus/procfs"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// see the documentation for /proc for more details about files and their format
+// https://www.kernel.org/doc/html/latest/filesystems/proc.html
+
+// openFilesLister stores the state needed to list open files
+// this is helpful for tests to help with mocking
+type openFilesLister struct {
+	pid      int
+	procPath string
+
+	readlink func(string) (string, error)
+	stat     func(string) (os.FileInfo, error)
+	lstat    func(string) (os.FileInfo, error)
+
+	proc       procfsProc
+	socketInfo map[uint64]socketInfo
+}
+
+// procfsProc is an interface to allow mocking of procfs.Proc
+type procfsProc interface {
+	ProcMaps() ([]*procfs.ProcMap, error)
+	FileDescriptors() ([]uintptr, error)
+	Cwd() (string, error)
+}
+
+type socketInfo struct {
+	Description string
+	State       string
+	Protocol    string
+}
+
+func openFiles(_ context.Context, pid int) (Files, error) {
+	ofl := &openFilesLister{
+		pid: pid,
+
+		readlink: os.Readlink,
+		stat:     os.Stat,
+		lstat:    os.Lstat,
```

Comment:
Why having this as part of the `openFilesLister`? Wouldn't work just calling `os.Stat` in the part that we needed it? 

Care to share more why you decided to go this path 😄

Reply:
I see you used to be able to mock things in tests. Very clever 👏

Reply:
Yep precisely, I initially didn't write it that way and testing was very painful, storing the syscall in the struct makes it way easier to test by mocking